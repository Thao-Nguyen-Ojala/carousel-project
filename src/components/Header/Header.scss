.header-wrapper{
  margin-top: 1.5rem;
  line-height: 2.8rem;

  &__title {
    font-size: 2.5rem;
    font-weight: 400;
    margin-left: auto;
  }

  &__restaurants-count {
    font-size: 1.5rem;
  }
}

@media screen and (max-width: 959px){
  .header-wrapper{
    &__title {
      font-size: 1.8rem;
      font-weight: 400;
      text-align: center;
      margin-left: auto;
    }
  
    &__restaurants-count {
      font-size: 1.3rem;
      text-align: center;
    }
  }
}