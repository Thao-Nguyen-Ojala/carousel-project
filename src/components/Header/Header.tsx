import './Header.scss';

type HeaderPropsType = {
  title: string;
  restaurantsCount: number;
};

const Header = ({ title, restaurantsCount }: HeaderPropsType) => {
  return (
    <div className="header-wrapper">
      <span>
        <div className="header-wrapper__title">{title}</div>
        <div className="header-wrapper__restaurants-count">
          {`(${restaurantsCount} restaurants)`}
        </div>
      </span>
    </div>
  );
};

export default Header;
