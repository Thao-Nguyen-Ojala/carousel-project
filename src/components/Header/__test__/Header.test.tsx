import React from 'react';
import ReactDOM from 'react-dom';
import Header from '../Header';
import { cleanup } from '@testing-library/react';
import '@testing-library/jest-dom';
import renderer from 'react-test-renderer';

afterEach(cleanup);

it('render Header without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<Header title="test1" restaurantsCount={1}></Header>, div);
});

it('matches Header snapshot', () => {
  const testingHeader = renderer
    .create(<Header title="Testing Header" restaurantsCount={3}></Header>)
    .toJSON();
  expect(testingHeader).toMatchSnapshot();
});
