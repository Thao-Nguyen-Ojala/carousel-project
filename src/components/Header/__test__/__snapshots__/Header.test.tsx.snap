// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches Header snapshot 1`] = `
<div
  className="header-wrapper"
>
  <span>
    <div
      className="header-wrapper__title"
    >
      Testing Header
    </div>
    <div
      className="header-wrapper__restaurants-count"
    >
      (3 restaurants)
    </div>
  </span>
</div>
`;
